ql/src/java/org/apache/hadoop/hive/ql/parse/ColumnStatsAutoGatherContext.java

```diff
@@ -71,13 +71,21 @@
   private Context origCtx;
   
   public ColumnStatsAutoGatherContext(SemanticAnalyzer sa, HiveConf conf,
-      Operator<? extends OperatorDesc> op, Table tbl, Map<String, String> partSpec,
+      Operator<? extends OperatorDesc> op, Table origTbl, Map<String, String> partSpec,
       boolean isInsertInto, Context ctx) throws SemanticException {
     super();
     this.sa = sa;
     this.conf = conf;
     this.op = op;
-    this.tbl = tbl;
+    try {
+      this.tbl = origTbl.copy();
+    } catch (HiveException he) {
+      throw new SemanticException(he);
+    }
+    if (tbl.getStorageHandler().alwaysUnpartitioned()) {
```

Comment:
Can the storage handler be null?

Reply:
Not sure, so I have added the `null` check